include/baresip.h

```diff
@@ -465,6 +465,14 @@ struct contact *contacts_current(const struct contacts *contacts);
 struct mediadev {
 	struct le   le;
 	char  *name;
+	struct {
+		int host_idx;
+		int dev_idx;
```

Comment:
host and device index is Portaudio specific.

This API is supposed to be generic...

Reply:
Maybe we can use `char *index` and `char *driver`, most audio APIs have some type of identifiers. Yes Portaudio is a bit special, since it provides multiple host drivers, but without populating these, its hard to tell the difference if a device is listed multiple times.

Reply:
perhaps split the API into generic and driver specific portions:

```c
/** Defines a media device */
struct mediadev {
	struct le   le;
	char  *name;

	/* Generic portion: */
	channels, default etc;
	...

	/* Module/driver specific: */
	int host_index;
	int device_index;
	...
};
```